python/private/config_settings.bzl

```diff
@@ -109,13 +115,29 @@ def construct_config_settings(*, name, default_version, versions, minor_mapping,
     # It's private because matching the concept of e.g. "3.8" value is done
     # using the `is_python_X.Y` config setting group, which is aware of the
     # minor versions that could match instead.
+    first_minor = None
     for minor in minor_mapping.keys():
+        if first_minor == None:
+            first_minor = minor
+
         native.config_setting(
             name = "is_python_{}".format(minor),
             flag_values = {_PYTHON_VERSION_MAJOR_MINOR_FLAG: minor},
             visibility = ["//visibility:public"],
         )
 
+    for minor in range(int(first_minor.partition(".")[-1]) + 1):
+        minor = "3.{}".format(minor)
+        if minor in minor_mapping:
+            break
+
+        # TODO @aignas 2025-11-04: use env-marker-setting with the smallest minor_mapping version
+        native.alias(
+            name = "is_python_{}".format(minor),
```

Comment:
We need targets that are for other Python versions, but maybe instead of this hack to make the versions incompatible, we colud do it differently:
```
for minor in range("3.0", max(minor_mapping)):
    native.config_setting()
```

That way we'll get the right behaviour. Another option would be to leave `minor_mapping[3.8] = None`. That way we might get the right behaviour as well.